### PR TITLE
Add configurable datetime object support

### PR DESCRIPTION
**Pull request by klatys**
- Bump nette php-generator version
- Drop php5.4 and php5.5 support
- Add php7.1 tests

### CODE REVIEW

**Comment by klatys** [comment]
Also added sf 4.0 as valid dependency

File: src/Skrz/Meta/DateTimeFormattingSerializer.php [hunk 1]
```diff
@@ -12,12 +12,22 @@ class DateTimeFormattingSerializer implements PropertySerializerInterface
...
-	public function __construct($format)
+	private $emptyValue;
+
+	public function __construct($format, $dateTimeClass = \DateTime::class, $emptyValue = '0000-00-00 00:00:00')
 	{
 		$this->format = $format;
+		$this->dateTimeClass = $dateTimeClass;
+		if (!substr($dateTimeClass, 0, 1) != "\\") {
+		    $this->dateTimeClass = "\\" . $this->dateTimeClass;
+        }
```
Comment (mzstic) [inline src/Skrz/Meta/DateTimeFormattingSerializer.php]: Indentation...

**Review by mzstic — COMMENTED** [review]
Except the indentation everyhting looks OK :)

**Comment by klatys** [comment]
my bad, fixed @mzstic

**Pull request merged**